das/daser.go

```diff
@@ -143,6 +158,14 @@ func (d *DASer) sample(ctx context.Context, sub header.Subscription, checkpoint
 			if err == context.Canceled {
 				return
 			}
+			var byzantineErr *ipld.ErrByzantine
+			if errors.As(err, &byzantineErr) {
+				err := d.fService.Broadcast(ctx, fraud.CreateBadEncodingProof(uint64(h.Height), byzantineErr))
+				if err != nil {
+					log.Errorw("fraud proof propagating failed with error ", err)
+				}
```

Comment:
Maybe to leave for a future PR, but shouldn't this be asynchronous?

Reply:
it could be async but I think it does not matted in this case, because we will stop sampling in case of befp.